upstream-sha256: d5c6abfb32fb971c3af20dc512be3b85b7686e26779ea493d31c0b6deb9fc38e
actual-sha256: 933552ab08513744a8238d826b67fdf0ccd396169c4852c48f4592bd3ba6c23f
--- upstream.expected
+++ tolk.actual
@@ -9,23 +9,27 @@
   float4 val1 = (*((device float4*)((data1_64+alu0))));
   float alu1 = -val0.x;
   float alu2 = ((alu1<-7.0f)?-7.0f:alu1);
-  float alu3 = -val0.z;
-  float alu4 = ((alu3<-7.0f)?-7.0f:alu3);
-  float alu5 = -val1.x;
-  float alu6 = ((alu5<-7.0f)?-7.0f:alu5);
-  float alu7 = -val1.z;
+  float alu3 = -alu2;
+  float alu4 = -val0.z;
+  float alu5 = ((alu4<-7.0f)?-7.0f:alu4);
+  float alu6 = -alu5;
+  float alu7 = -val1.x;
   float alu8 = ((alu7<-7.0f)?-7.0f:alu7);
-  float alu9 = ((val0.y<-7.0f)?-7.0f:val0.y);
-  float alu10 = -alu9;
+  float alu9 = -alu8;
+  float alu10 = -val1.z;
   float alu11 = ((alu10<-7.0f)?-7.0f:alu10);
-  float alu12 = ((val0.w<-7.0f)?-7.0f:val0.w);
-  float alu13 = -alu12;
-  float alu14 = ((alu13<-7.0f)?-7.0f:alu13);
-  float alu15 = ((val1.y<-7.0f)?-7.0f:val1.y);
+  float alu12 = -alu11;
+  float alu13 = ((val0.y<-7.0f)?-7.0f:val0.y);
+  float alu14 = -alu13;
+  float alu15 = ((val0.w<-7.0f)?-7.0f:val0.w);
   float alu16 = -alu15;
-  float alu17 = ((alu16<-7.0f)?-7.0f:alu16);
-  float alu18 = ((val1.w<-7.0f)?-7.0f:val1.w);
-  float alu19 = -alu18;
-  float alu20 = ((alu19<-7.0f)?-7.0f:alu19);
-  *((device float4*)((data0_32+(lidx0<<2)))) = float4(-(alu6*(1/(exp2((alu6*2.4554669857025146f))+1.0f))*(1.0f-alu17)),-(alu8*(1/(exp2((alu8*2.4554669857025146f))+1.0f))*(1.0f-alu20)),-(alu2*(1/(exp2((alu2*2.4554669857025146f))+1.0f))*(1.0f-alu11)),-(alu4*(1/(exp2((alu4*2.4554669857025146f))+1.0f))*(1.0f-alu14)));
+  float alu17 = ((val1.y<-7.0f)?-7.0f:val1.y);
+  float alu18 = -alu17;
+  float alu19 = ((val1.w<-7.0f)?-7.0f:val1.w);
+  float alu20 = -alu19;
+  float alu21 = ((alu14<-7.0f)?-7.0f:alu14);
+  float alu22 = ((alu16<-7.0f)?-7.0f:alu16);
+  float alu23 = ((alu18<-7.0f)?-7.0f:alu18);
+  float alu24 = ((alu20<-7.0f)?-7.0f:alu20);
+  *((device float4*)((data0_32+(lidx0<<2)))) = float4((alu9*(1/(exp2((alu9*1.7020000219345093f*-1.4426950216293335f))+1.0f))*(1.0f-alu23)),(alu12*(1/(exp2((alu12*1.7020000219345093f*-1.4426950216293335f))+1.0f))*(1.0f-alu24)),(alu3*(1/(exp2((alu3*1.7020000219345093f*-1.4426950216293335f))+1.0f))*(1.0f-alu21)),(alu6*(1/(exp2((alu6*1.7020000219345093f*-1.4426950216293335f))+1.0f))*(1.0f-alu22)));
 }
